### ISSUE TITLE
gofumpt false positive when used with goimports local-prefixes

### ISSUE THREAD

**Issue by strive1216** [golangci/golangci-lint#6266]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Brew

### Description of the problem

When both goimports (configured with local-prefixes) and gofumpt are enabled, golangci-lint run incorrectly reports a gofumpt formatting error (File is not properly formatted (gofumpt)).

However, running golangci-lint run --fix on the same file reports 0 issues, and running gofumpt -d directly on the file also passes. This suggests a conflict or a reporting issue within golangci-lint's orchestration of these two linters, rather than an actual formatting problem with the file itself.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 2.7.2 built with go1.25.4 from 9f61b0f5 on 2025-12-07T16:57:12Z
```

</details>

### Configuration

<details>

```console
version: "2"
linters:
   ........
formatters:
  enable:
    - gofumpt
    - goimports
    - golines
  settings:
    goimports:
      local-prefixes:
        - server
    gofumpt:
      extra-rules: true
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.25.5 darwin/arm64
AR='ar'
CC='clang'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='clang++'
GCCGO='gccgo'
GO111MODULE='on'
GOARCH='arm64'
GOARM64='v8.0'
GOAUTH='netrc'
GOBIN=''
GOCACHE='/Users/strive/Library/Caches/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/Users/strive/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/w_/3v0vbqgd0kn1d_03rv4zk54h0000gn/T/go-build642541099=/tmp/go-build -gno-record-gcc-switches -fno-common'
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMOD='/dev/null'
GOMODCACHE='/Users/strive/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/strive/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/Users/strive/.go/go1.25.5'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/strive/Library/Application Support/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/Users/strive/.go/go1.25.5/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.25.5'
GOWORK=''
PKG_CONFIG='pkg-config'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO golangci-lint has version 2.7.2 built with go1.25.4 from 9f61b0f5 on 2025-12-07T16:57:12Z 
INFO [config_reader] Config search paths: [./ /Users/strive/Work/server/common /Users/strive/Work/server /Users/strive/Work /Users/strive /Users /] 
INFO [config_reader] Used config file ../../../.golangci.yml 
INFO [config_reader] Module name "server"         
INFO maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined 
INFO [goenv] Read go env for 7.316584ms: map[string]string{"GOCACHE":"/Users/strive/Library/Caches/go-build", "GOROOT":"/Users/strive/.go/go1.25.5"} 
INFO [lintersdb] Active 10 linters: [errcheck gocyclo gofumpt goimports golines gosec govet ineffassign staticcheck unused] 
INFO [loader] Go packages loading at mode 8767 (types_sizes|compiled_files|exports_file|imports|deps|files|name) took 456.5435ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 658.125µs 
INFO [linters_context/goanalysis] analyzers took 33.046745181s with top 10 stages: buildir: 25.861734342s, fact_deprecated: 1.261030038s, inspect: 1.239820252s, printf: 1.08987025s, ctrlflow: 972.670909ms, fact_purity: 710.730331ms, nilness: 635.013904ms, typedness: 561.651252ms, SA5012: 450.998991ms, tokenfileanalyzer: 179.921993ms 
INFO [runner] Issues before processing: 2, after processing: 1 
INFO [runner] Processors filtering stat (in/out): max_from_linter: 1/1, path_prettifier: 1/1, path_relativity: 2/2, generated_file_filter: 2/2, exclusion_rules: 2/2, nolint_filter: 2/2, fixer: 2/2, source_code: 1/1, filename_unadjuster: 2/2, exclusion_paths: 2/2, diff: 2/2, max_same_issues: 1/1, severity-rules: 1/1, uniq_by_line: 2/2, max_per_file_from_linter: 2/1, path_absoluter: 2/2, cgo: 2/2, invalid_issue: 2/2, path_shortener: 1/1, sort_results: 1/1 
INFO [runner] processing took 395.207µs with stages: nolint_filter: 296.584µs, generated_file_filter: 47.75µs, source_code: 32.126µs, path_relativity: 6.208µs, uniq_by_line: 3.541µs, sort_results: 3.208µs, max_same_issues: 1.041µs, max_per_file_from_linter: 958ns, path_shortener: 625ns, invalid_issue: 624ns, max_from_linter: 500ns, cgo: 459ns, fixer: 333ns, path_absoluter: 291ns, exclusion_paths: 209ns, path_prettifier: 167ns, diff: 167ns, filename_unadjuster: 167ns, exclusion_rules: 125ns, severity-rules: 124ns 
INFO [runner] linters took 7.384562417s with stages: goanalysis_metalinter: 7.384110542s 
server/common/setup.go:5:1: File is not properly formatted (gofumpt)
        "time"
^
1 issues:
* gofumpt: 1
INFO File cache stats: 1 entries of total size 782B 
INFO Memory: 80 samples, avg is 812.2MB, max is 1245.3MB 
INFO Execution took 7.849651917s
$ golangci-lint run -v --fix
INFO golangci-lint has version 2.7.2 built with go1.25.4 from 9f61b0f5 on 2025-12-07T16:57:12Z 
INFO [config_reader] Config search paths:  [./ /Users/strive/Work/server/common /Users/strive/Work/server /Users/strive/Work /Users/strive /Users /] 
INFO [config_reader] Used config file ../../../.golangci.yml 
INFO [config_reader] Module name "server"         
INFO maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined 
INFO [goenv] Read go env for 6.303416ms: map[string]string{"GOCACHE":"/Users/strive/Library/Caches/go-build", "GOROOT":"/Users/strive/.go/go1.25.5"} 
INFO [lintersdb] Active 10 linters: [errcheck gocyclo gofumpt goimports golines gosec govet ineffassign staticcheck unused] 
INFO [loader] Go packages loading at mode 8767 (types_sizes|compiled_files|name|deps|exports_file|files|imports) took 247.805666ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 159.041µs 
INFO [linters_context/goanalysis] analyzers took 0s with no stages 
INFO [runner] Applying suggested fixes            
INFO [runner] fixer took 8.40375ms with stages: all: 8.40375ms 
INFO [runner] Issues before processing: 2, after processing: 0 
INFO [runner] Processors filtering stat (in/out): filename_unadjuster: 2/2, path_relativity: 2/2, generated_file_filter: 2/2, diff: 2/2, path_absoluter: 2/2, exclusion_paths: 2/2, cgo: 2/2, invalid_issue: 2/2, exclusion_rules: 2/2, nolint_filter: 2/2, fixer: 2/0 
INFO [runner] processing took 8.562792ms with stages: fixer: 8.439208ms, nolint_filter: 89.5µs, generated_file_filter: 25.708µs, path_relativity: 2.875µs, sort_results: 2.167µs, invalid_issue: 542ns, max_same_issues: 458ns, cgo: 417ns, exclusion_rules: 250ns, path_absoluter: 249ns, uniq_by_line: 209ns, exclusion_paths: 209ns, path_prettifier: 208ns, max_from_linter: 166ns, path_shortener: 166ns, filename_unadjuster: 125ns, max_per_file_from_linter: 84ns, diff: 84ns, severity-rules: 84ns, source_code: 83ns 
INFO [runner] linters took 169.882292ms with stages: goanalysis_metalinter: 161.254458ms 
0 issues.
INFO File cache stats: 1 entries of total size 782B 
INFO Memory: 6 samples, avg is 49.3MB, max is 63.4MB 
INFO Execution took 424.847958ms      

$ gofumpt -d ./setup.go

```
</details>

### A minimal reproducible example or link to a public repository

<details>

```go
package common

import (
	"flag"
	"time"

	"server/common/cache"
	"server/common/xgorm"
	"server/internal/config"
	"server/model"

	"github.com/zeromicro/go-zero/core/conf"
	"github.com/zeromicro/go-zero/core/logx"
	"gorm.io/driver/postgres"
	"gorm.io/gorm"
)
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [x] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
This is because you are using an invalid module name: `module server`.

Non-namespaced module names are reserved for Go itself.

Those module names should not be use.

https://go.dev/ref/mod#module-path
